taraxa/state/state_db_rocksdb/db.go

```diff
@@ -113,145 +115,131 @@ func (self *DB) Snapshot(dir string, log_size_for_flush uint64) error {
 	return c.CreateCheckpoint(dir, log_size_for_flush)
 }
 
-func (self *DB) Prune(state_root_to_keep common.Hash, state_root_to_prune []common.Hash, blk_num types.BlockNum) {
+func (self *DB) RecreateColumn(column state_db.Column, nodes map[common.Hash][]byte) {
+	range_limit := [common.VersionedKeyLength]byte{255}
+	range32 := grocksdb.Range{
+		Start: make([]byte, common.HashLength),
+		Limit: make([]byte, common.HashLength),
+	}
+	copy(range32.Limit, range_limit[0:common.HashLength])
+	err := self.db.DropColumnFamily(self.cf_handles[column])
+	if err != nil {
+		panic("Could not delete column")
+	}
+	self.cf_handles[column].Destroy()
+	acc_trie_handle, err2 := self.db.CreateColumnFamily(self.db_opts, strconv.Itoa(int(column+1)))
+	if err2 != nil {
+		panic("Could not create column")
+	}
+	self.cf_handles[column] = acc_trie_handle
+
+	for node_to_keep, b := range nodes {
+		self.db.PutCF(grocksdb.NewDefaultWriteOptions(), self.cf_handles[column], node_to_keep[:], b)
+	}
+	self.db.CompactRangeCF(self.cf_handles[column], range32)
+}
+
+func (self *DB) Prune(state_root_to_keep []common.Hash, blk_num types.BlockNum) {
```

Comment:
Maybe it will be better to split this method to smaller ones related to some specific column?
Feel free to resolve it I am missing something and its not possible

Reply:
Matus is workin on task to parallelize some of these things inside of this method so I would defer this to that task